shared/components/TrackEventWhenLoaded/TrackEventWhenLoaded.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { useAuthState } from 'shared/providers/AuthProvider';
+import amplitude from 'shared/lib/amplitude';
+
+const objectToGqlInput = (props = {}) =>
+  Object.keys(props).map((key) => ({
+    field: key,
+    value: props[key],
+  }));
+
+const TrackEventWhenLoaded = ({ loaded, eventName, properties }) => {
+  const { authenticated } = useAuthState();
+
+  useEffect(() => {
+    if (loaded && authenticated) {
+      amplitude.trackEvent({
+        eventName,
+        properties: objectToGqlInput(properties),
+      });
+    }
+  }, [loaded, authenticated, properties, eventName]);
+
+  return null;
+};
+
+TrackEventWhenLoaded.propTypes = {
+  loaded: PropTypes.bool,
+  eventName: PropTypes.string.isRequired,
+  // eslint-disable-next-line react/forbid-prop-types
+  properties: PropTypes.any,
```

Comment:
Not critical by any means, but, it looks like only `objects` are acceptable since we pipe it through `Object.keys` before sending to GQL, right?

```suggestion
  properties: PropTypes.object,
```